charts/karpenter/templates/controller/rbac.yaml

```diff
@@ -40,24 +40,27 @@ rules:
 - apiGroups: [""]
   resources: ["events"]
   verbs: ["create"]
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
+  verbs: ["create", "get", "patch", "update", "watch"]
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: karpenter-controller
 rules:
 - apiGroups: ["karpenter.sh"]
-  resources: ["provisioners", "provisioners/status"]
+  resources: ["provisioners"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["karpenter.sh"]
+  resources: ["provisioners/status"]
   verbs: ["create", "delete", "patch", "get", "list", "watch"]
-- apiGroups: ["coordination.k8s.io"]
-  resources: ["leases"]
-  verbs: ["create", "get", "patch", "update", "watch"]
 - apiGroups: [""]
   resources: ["nodes", "pods"]
   verbs: ["get", "list", "watch", "patch", "delete"]
 - apiGroups: [""]
   resources: ["configmaps"]
-  verbs: ["get", "list", "watch", "update"]
```

Comment:
I *think* that knative/pkg's configmap controller might need this to update configmaps. We should be able to restrict it to the karpenter namespace.

Reply:
Oh I see, this is the controller's rbac.

Reply:
What do you have in mind as it relates to approving the running workflow?

No rush, just curious to see the workflow run to completion and getting acquainted to the contribution process for this project.

Either way - this should be the first contribution of many since I plan to consistently contribute to this project going forward.

Reply:
> Either way - this should be the first contribution of many since I plan to consistently contribute to this project going forward.

Awesome! 

Unfortunately, RBAC is not included in the current testing workflow. We run `make battletest` as part of CI, which stands up a local APIServer/ETCD using controller-runtime's `testenv` package. In order to test this, you should run `make apply` (see https://karpenter.sh/docs/development-guide/) and do some basic scaling and provisioner CRUD. 

We definitely need to improve this -- I'll cut a ticket.

Reply:
Were you able to test this manually?